src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -602,46 +657,15 @@ function discoverController(
         })
       );
 
-      $scope.$watchCollection('state.sort', function(sort) {
-        if (!sort) return;
-
-        // get the current sort from searchSource as array of arrays
-        const currentSort = getSortArray($scope.searchSource.getField('sort'), $scope.indexPattern);
-
-        // if the searchSource doesn't know, tell it so
-        if (!angular.equals(sort, currentSort)) $fetchObservable.next();
-      });
-
-      // update data source when filters update
-
-      subscriptions.add(
-        subscribeWithScope($scope, filterManager.getUpdates$(), {
-          next: () => {
-            $scope.updateDataSource().then(function() {
-              $state.save();
-            });
-          },
-        })
-      );
-
-      // update data source when hitting forward/back and the query changes
-      $scope.$listen($state, 'fetch_with_changes', function(diff) {
-        if (diff.indexOf('query') >= 0) $fetchObservable.next();
-      });
-
-      $scope.$watch('opts.timefield', function(timefield) {
-        $scope.enableTimeRangeSelector = !!timefield;
-      });
-
       $scope.$watch('state.interval', function(newInterval, oldInterval) {
         if (newInterval !== oldInterval) {
-          $fetchObservable.next();
+          setAppState({ interval: newInterval });
```

Comment:
Why not set the state directly in the correct `subscribeWithScope` and get rid of `state.interval` and of `$scope.refreshInterval` all together?

Reply:
added a comment to point out that's the histogramInterval in this case